### PR TITLE
[Feature] Configure tolerations in kubernetes backend

### PR DESCRIPTION
**Pull request by klinux**
This code add a feature to support tolerations in Kubernetes Backend

### CODE REVIEW

**Comment by 6543** [comment]
you have to alter the TestCompilerCompile test too :D ... if you need help just ask and I'll push

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 1]
```diff
@@ -84,6 +84,11 @@ Specify the label which is used to select the node where the job should be execu
 By default the pod will use "kubernetes.io/arch" inferred from top-level "platform" setting which is deducted from the agents' environment variable CI_SYSTEM_PLATFORM. To overwrite this, you need to specify this label in the nodeSelector section.
 See the [kubernetes documentation](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) for more information on using nodeSelector.
 
+### tolerations
+
+When you use nodeSelector and the node pool is configured with Taints, you needs to specify the Tollerations. Tolerations allow the scheduler to schedule pods with matching taints.
```
Comment (zc-devs) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:89]:
```suggestion
When you use nodeSelector and the node pool is configured with Taints, you need to specify the Tolerations. Tolerations allow the scheduler to schedule pods with matching taints.
```

File: pipeline/backend/types/backend_kubernetes.go [hunk 2]
```diff
@@ -19,10 +19,35 @@ type KubernetesBackendOptions struct {
...
+	Tolerations        []Toleration      `json:"tolerations,omitempty"`
 }
 
 // Resources defines two maps for kubernetes resource definitions
 type Resources struct {
 	Requests map[string]string `json:"requests,omitempty"`
 	Limits   map[string]string `json:"limits,omitempty"`
 }
+
+// Tolerations defines an array with kubernetes tolerations definition
```
Comment (zc-devs) [inline pipeline/backend/types/backend_kubernetes.go:31]:
```suggestion
// Defines Kubernetes toleration definition
```

File: pipeline/backend/kubernetes/pod.go [hunk 3]
```diff
@@ -125,6 +125,20 @@ func Pod(namespace string, step *types.Step, labels, annotations map[string]stri
...
 
+	var tolerations []v1.Toleration
+	if len(step.BackendOptions.Kubernetes.Tolerations) > 0 {
+		for _, t := range step.BackendOptions.Kubernetes.Tolerations {
+			toleration := v1.Toleration{
+				Key:               t.Key,
+				Operator:          v1.TolerationOperator(t.Operator),
+				Value:             t.Value,
+				Effect:            v1.TaintEffect(t.Effect),
+				TolerationSeconds: t.TolerationSeconds,
```
Comment (zc-devs) [inline pipeline/backend/kubernetes/pod.go:136]: Should we add this in docs (Example pipeline configuration)?

File: pipeline/backend/kubernetes/pod.go [hunk 4]
```diff
@@ -136,6 +150,7 @@ func Pod(namespace string, step *types.Step, labels, annotations map[string]stri
 			RestartPolicy:      v1.RestartPolicyNever,
 			HostAliases:        hostAliases,
 			NodeSelector:       nodeSelector,
+			Tolerations:        tolerations,
```
Comment (zc-devs) [inline pipeline/backend/kubernetes/pod.go:153]: And maybe logging (trace) actual values which pod is going to run with is good idea (see node selector logging).

Comment (klinux) [reply pipeline/backend/kubernetes/pod.go:155]: Can you check the implementatiion?

**Comment by klinux** [comment]
@6543 sure, can you help me with that? Thank you

File: pipeline/backend/kubernetes/pod.go (hunk 4 above)
Comment (zc-devs) [reply pipeline/backend/kubernetes/pod.go:155]: Yeah, can test it tomorrow.

**Review by zc-devs — APPROVED** [review]

**Comment by 6543** [comment]
well looks like you did got it without my help 😅 🎉

**Review by 6543 — APPROVED** [review]

**Pull request merged**